src/modules/wishlists/wishlists.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Accommodation } from '#/modules/accommodations/entities/accommodations.entity';
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Wishlist } from '#/modules/wishlists/entities/wishlist.entity';
+
+@Injectable()
+export class WishlistService {
+  constructor(
+    @InjectRepository(Wishlist)
+    private readonly wishlistRepository: Repository<Wishlist>,
+    @InjectRepository(Accommodation)
+    private readonly accommodationRepository: Repository<Accommodation>,
+  ) {}
+
+  async addToWishlist(accountId: string, accommodationId: string) {
+    const existingWishlistItem = await this.wishlistRepository.findOne({
+      where: { accountId, accommodationId },
+    });
+    if (existingWishlistItem) {
+      throw new ConflictException('Wishlist item already exists');
+    }
+    const accommodation = await this.accommodationRepository.findOne({
+      where: { id: accommodationId },
+    });
+    if (!accommodation) {
+      throw new NotFoundException('Accomodation not found');
+    }
+
+    const wishlistItem = this.wishlistRepository.create({
+      accountId,
+      accommodationId,
+    });
+
+    const savedWish = await this.wishlistRepository.save(wishlistItem);
+    return savedWish;
+  }
+
+  async getUserWishlist(accountId: string) {
+    const userWishlist = await this.wishlistRepository.find({
+      where: { accountId },
+    });
+
+    return userWishlist;
```

Comment:
Also, please return the actual accommodation itself instead of just its ```id```. In FE, we will need it to display cards

Reply:
Can't understand why it's needed. I mean, if I will return accommodation it will be same thing as it was, I did no changes on it.

Reply:
I can return but I can't imagine how you are gonna to use it

Reply:
> Can't understand why it's needed. I mean, if I will return accommodation it will be same thing as it was, I did no changes on it.

Now you're returning the wishlist items only. You should return accommodation relation.
You should add ```relations: ['accommodation']``` to the query

Reply:
Oh I fix that, only then I understand what you mean

Reply:
> I can return but I can't imagine how you gonna to use it

This is the airbnb wishlist:
![image](https://github.com/user-attachments/assets/9094acc8-9193-4f12-a716-f25b2482e751)
In order to display such card, we need the accommodation details, right?

Reply:
> Oh I fix that, only then I understand what you mean

Already got you :)